### PR TITLE
Network NG: Move Interfaces to InterfacesCollection

### PR DESCRIPTION
**Pull request by imobachgs**
* Move `Y2Network::Interfaces` is moved to `Y2Network::InterfacesCollection`.
* Add unit tests.
* Move interfaces reading to `ConfigReader::Sysconfig`.

### CODE REVIEW

File: src/lib/y2network/config_reader/sysconfig.rb [hunk 1]
```diff
@@ -66,6 +65,18 @@ def find_routing_tables(interfaces)
...
+      # Returns the list of network interfaces
+      #
+      # @return [Array<Interface>] Interfaces
+      def find_interfaces
+        # LanItems has to be already initialized
+        interfaces = Yast::LanItems.Items.map do |_index, item|
+          name = item["ifcfg"] || item["hwinfo"]["dev_name"]
+          Y2Network::Interface.new(name)
+        end
+        Y2Network::Interfaces.new(interfaces)
```
Comment (teclator) [inline src/lib/y2network/config_reader/sysconfig.rb]:
```suggestion
        Y2Network::InterfacesCollection.new(interfaces)
```

File: src/modules/LanItems.rb [hunk 2]
```diff
@@ -1553,7 +1553,7 @@ def GetItemDescription
     end
 
     # Select the hardware component
-    # @param hardware the component
+    cpf# @param iface [Y2Network::Interface] Network interface
```
Comment (teclator) [inline src/modules/LanItems.rb]:
```suggestion
    # @param iface [Y2Network::Interface] Network interface
```

File: src/modules/LanItems.rb [hunk 3]
```diff
@@ -1320,7 +1320,7 @@ def bridge_slaves(master)
 
     # Creates item's startmode human description
     #
-    # @param item_id [Integer] a key for {#Items}
+    # @param ifcfg [Hash] Interface sysconfig configuration
```
Comment (mchf) [inline src/modules/LanItems.rb]: consistency note. Parameter carrying device's sysconfig is usually named ```devmap```. ```ifcfg``` is usually used for configuration name (```ifcfg == eth0``` --> file with sysconfig is ```ifcfg-eth0```)

**Comment by teclator** [comment]
LGTM

**Comment by mchf** [comment]
lgtm

**Pull request merged**